BridgeCareApp/BridgeCareCodeFirst/Models/BudgetReportData.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Linq;
+using static BridgeCare.Models.BudgetReportData;
+
+namespace BridgeCare.Models
+{
+    public class BudgetReportData : IBudgetReportData
+    {
+        private readonly List<ICostDetails> costList;
+
+        public BudgetReportData()
+        {
+        }
+        public BudgetReportData(List<ICostDetails> costList)
+        {
+            this.costList = costList ?? throw new ArgumentNullException(nameof(costList));
+        }
+        public BudgetReportDetails GetBudgetReportData(SimulationResult data, string[] budgetTypes)
+        {
+            var BudgetYearView = new Hashtable();
+            var db = new BridgeCareContext();
```

Comment:
Test to see if this can be injected. May impact performance.

Reply:
Injecting DbContext in all the places but one, where heavy data operation is going on.
The performance reduction is not significant.